superset/utils/log.py

```diff
@@ -352,3 +355,71 @@ def log(  # pylint: disable=too-many-arguments,too-many-locals
         except SQLAlchemyError as ex:
             logging.error("DBEventLogger failed to log event(s)")
             logging.exception(ex)
+
+
+class AsyncPruneEventScheduleLogCommand(AbstractEventLogger):
+    """
+    Prunes events logs
+    """
+
+    def __init__(self, worker_context: bool = True):
+        self._worker_context = worker_context
+
+    def run(self) -> None:
+
+        # pylint: disable=import-outside-toplevel
+        from superset.utils.celery import session_scope
+
+        log_retention = current_app.config["EVENT_LOG_PRUNE"]
+        
+        with session_scope(nullpool=True) as session:
+            self.validate()
+            prune_errors = []
+
+            if log_retention is not None:
+                from_date = datetime.utcnow() - timedelta(days=log_retention)
+                try:
+                    row_count = bulk_delete_logs(
+                        from_date, session=session, commit=False
+                    )
+                    logging.info(
+                        "Deleted %s logs for events",
+                        str(row_count),
+                    )
+                except DAODeleteFailedError as ex:
+                    prune_errors.append(str(ex))
+            if prune_errors:
+                raise "An error occurred while pruning events logs;".join(prune_errors)
+
+    def validate(self) -> None:
+        pass
+
+
+def bulk_delete_logs(
```

Comment:
Move this method into the `Command` It's not really pure as it relies on `db.session`, which is a proxy

Reply:
He is following the same pattern that already exists, I'de vote for keeping it here